homeassistant/components/aidot/config_flow.py

```diff
@@ -0,0 +1,181 @@
+        self.__login_control = LoginControl()
+        self.login_response: dict[Any, Any] = {}
+        self.accessToken = ""
+        self.house_list: list[Any] = []
+        self.device_list: list[Any] = []
+        self.product_list: list[Any] = []
+        self.selected_house: dict[Any, Any] = {}
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                # get ContryCode
+                selected_contry_name = user_input[CONF_COUNTRY]
+                self.__login_control.change_country_name(selected_contry_name)
+
+                self.login_response = await self.__login_control.async_post_login(
+                    self.hass,
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+                if self.login_response is None:
+                    errors["base"] = "login_failed"
+                    return self.async_show_form(
+                        step_id="user", data_schema=None, errors=errors
+                    )
+                self.accessToken = self.login_response["accessToken"]
+
+                # get houses
+                self.house_list = await self.__login_control.async_get_houses(
+                    self.hass, self.accessToken
+                )
+                if self.house_list is None or len(self.house_list) == 0:
+                    errors["base"] = "get_house_failed"
+                    return self.async_show_form(
+                        step_id="user", data_schema=None, errors=errors
+                    )
+
+                return await self.async_step_choose_house()
+
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidHost:
+                errors["host"] = "cannot_connect"
+        if user_input is None:
+            user_input = {}
+
+        DATA_SCHEMA = vol.Schema(
+            {
+                vol.Required(
+                    CONF_COUNTRY,
+                    default=user_input.get(CONF_COUNTRY, DEFAULT_COUNTRY_NAME),
+                ): vol.In(SUPPORTED_COUNTRY_NAMES),
+                vol.Required(
+                    CONF_USERNAME, default=user_input.get(CONF_USERNAME, vol.UNDEFINED)
+                ): str,
+                vol.Required(
+                    CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, vol.UNDEFINED)
+                ): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_choose_house(self, user_input=None) -> ConfigFlowResult:
+        """Please select a room."""
+        errors: dict[str, str] = {}
+        if user_input is None:
+            user_input = {}
+
+        if user_input.get(CONF_CHOOSE_HOUSE) is not None:
```

Comment:
```suggestion
        if user_input is not None:
```
it's a required field anyway